src/ota_image_libs/v1/file_table/db.py

```diff
@@ -305,33 +310,30 @@ def iter_dir_entries(self) -> Generator[DirTypedDict]:
             )
             # fmt: on
 
-    def iter_regular_entries(self) -> Generator[RegularFileTypedDict]:
+    def iter_regular_entries(self) -> Generator[RegularFileRow]:
```

Comment:
Seems this is API. At least, will be better to verify input/output in the test code.
NOTE: I know you said you want to increase test coverage, maybe this is one of them.